RecipesManagerApi.Infrastructure/Services/RecipesService.cs

```diff
@@ -153,6 +157,8 @@ public async Task<RecipeDto> GetRecipeAsync(string id, CancellationToken cancell
             throw new InvalidDataException("Provided id is invalid.");
         }
         var entity = await this._recipesRepository.GetRecipeAsync(objectId, GlobalUser.Id.Value, cancellationToken);
+        var viewDto = new RecipeViewActivityDto{RecipeId = objectId};
+        await this._userActivityService.AddRecipeViewActivityAsync(viewDto, cancellationToken);
```

Comment:
Run in separate thread using Task.Run() because currently this method blocks main thread and slows down getting a recipe